update structurizr to 3.0.0

jgenssler-GiN opened the pull request:
This PR updates Structurizr to the latest working version. Attempting to upgrade beyond this version causes build failures.

The main motivation for this update is to support the `group { ... }` syntax in our models.

`ExportStructurizrSpec` test passed. However, the full test suite failed on my local environment (needs investigation).

### All Submissions:

* [x] Did you update the `changelog.adoc`?
* [ ] Does your PR affect the documentation?
* [ ] If yes, did you update the documentation or create an issue for updating it?

The source of the documentation can be found in `/src/docs/`.

If you didn't find the time to update docs, please create an issue as reminder to do so.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Your first submission

* [ ] Welcome to the list of contributors! If you have any questions, feel free to ask them by creating a new issue
* [x] Have you added your name to the list of [30_community.adoc](https://github.com/docToolchain/docToolchain/blob/ng/src/docs/10_about/30_community.adoc)?


inspiration: https://github.com/stevemao/github-issue-templates

rdmueller reviewed (approved):
LGTM

rdmueller commented:
Thank you!

rdmueller commented:
hm. all builds run into problems. Can you please take a second look?

ghost commented:
The builds for Java 11 and 14 were failing, which seems to be due to Structurizr now targeting Java 17 (see [this commit](https://github.com/structurizr/java/commit/940a0e9b2533c5017c9fad9fef11e30a065af9f7)).

To get the build passing, I dropped support for Java 11 and Java 14. However, since I’m not a Java developer, I’m unsure if this is the best long-term solution.

If this approach isn’t suitable for you, feel free to close this PR without merging.

Commit:
Drop support for Java 11 and 14

Structurizr depends on Java 17.

Signed-off-by: Jakob Genßler <jakob.genssler@gin.de>

Commit:
Update structurizr to 3.0.0

Bump structurizr to latest working version.
Increasing them further breaks the build.

Signed-off-by: Jakob Genßler <jakob.genssler@gin.de>
